src/map.h

```diff
@@ -161,6 +161,105 @@ struct level_cache {
 };
 
 /**
+ * Base of the map class, currently rather underpopulated.
+ * Is NOT supposed to implement complex behaviors, only an interface to access submaps.
+ */
+class map_base
+{
+    protected:
+        map_base( int mapsize = MAPSIZE, bool zlev = false );
+        map_base( bool zlev ) : map_base( MAPSIZE, zlev ) { }
+        ~map_base();
+
+        int my_MAPSIZE;
+        bool zlevels;
+
+        /**
+         * Absolute coordinates of first submap (get_submap_at(0,0))
+         * This is in submap coordinates (see overmapbuffer for explanation).
+         * It is set upon:
+         * - loading submap at grid[0],
+         * - generating submaps (@ref generate)
+         * - shifting the map with @ref shift
+         */
+        tripoint abs_sub;
+        /**
+         * Sets @ref abs_sub, see there. Uses the same coordinate system as @ref abs_sub.
+         */
+        void set_abs_sub( const int x, const int y, const int z );
```

Comment:
While you change this anyway, you could remove the top-level `const` (like here) as it has meaning in a function *declaration*.